### PR TITLE
Adding plugins to the community list

### PR DESCRIPTION
**Pull request by piercus**

### CODE REVIEW

File: docs/extending/plugins-list.md [hunk 1]
```diff
@@ -34,3 +34,11 @@
   - [fail](https://github.com/semantic-release/exec#fail): Execute a shell command to notify of a failed release
 
 ## Community plugins
+
+**[Add your plugin to the list](https://github.com/semantic-release/semantic-release/edit/caribou/docs/extending/plugins-list.md)**
+
+ - [semantic-release-docker](https://github.com/felixfbecker/semantic-release-docker) - Set of semantic-release plugins for publishing a docker image to Docker Hub.
```
[the remark this answers is not in the document]
Comment (piercus) [reply docs/extending/plugins-list.md]: I have contacted @felixfbecker to get his approval about adding him in the list

**Comment by piercus** [comment]
Hello @pvdlg, 

Thank you for your complete review.

I'd like to suggest another thing :-)
 
Extended plugin doc could promote a community npm keyword (like `semantic-release-plugin`) and just link to [semantic-release-plugin on npm](https://www.npmjs.com/search?q=keywords%3Asemantic-release-plugin) with this keyword. 

So no maintenance needed in the documentation, and no responsability on semantic-release's team about non-compatible plugins.

What do you think ?

File: docs/extending/plugins-list.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (piercus) [reply docs/extending/plugins-list.md]: I've contacted the author, he can only guarantee that it is working for ^11.0.0, but he would be happy to be listed.

**Comment by piercus** [comment]
@pvdlg I'm feeling confused, there is only mine remaining, i did really want to do it for the community :-(

File: docs/extending/plugins-list.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (pvdlg) [reply docs/extending/plugins-list.md]: Ok we can keep the link then.

[the remark this answers is not in the document]
Comment (piercus) [reply docs/extending/plugins-list.md]: Ok done

**Comment by pvdlg** [comment]
Plugins were implemented in semantic-release to allow a separation of concern (the plugin do stuff specific to a platform, the core do generic things) and to allow to support specific package managers.

Contrary to project like Gulp or Grunt, that are task runners which can do virtually anything, semantic-release has a much more limited scope: releasing packages. There is a finite number of Git hosted platform, package manager registry, commit conventions and release note format to support.

Even the `verifyRelease` step has a finite number of application as it's useful only for verifications related to a release. It's useful when you want to check the release version/type is correct, but if you make verification unrelated to the release version then they should be done as part of your test suite (before running semantic-release).
For example in your plugin `semantic-release-verify-deps` I would have recommended you to do that as part of your tests suite, so the build of PRs that contains local dependencies would fail, and that would avoid to merge it.
I'm totally ok to list the plugin though. I'm just mentioning it's not necessary always the best approach.

As a general rules, the earlier you catch a problem the better.
Problems that can be caught in the test phase should be handled there. A few example of things that should be done in test:
- Verify if dependencies have security vulnerabilities
- Verify that certain type of dependencies are not used
- Verify code format (linting)

A few example of things that make sense to do as a `verifyRelease` step:
- Verify that if the code break dependents, the release is a `major`
- Verify that if the public API changed the release is at least a `minor`
- Verify that if the integration tests changed, the release is a `major`

In any case, I don't think the list of plugin for semantic-release is meant to grow to more than a handful or a dozen plugins. Especially because the "Official" ones already cover a lot of use cases.
In addition, most scenario can be implemented with the [@semantic-release/exec](https://github.com/semantic-release/exec).

Therefore I don't think having an autogenerated list based on an npm keyword would be necessary.

Regarding the plugin we include in the list, it's a lot more important to have well maintained plugins rather than having a long list. Here is a  few reasons:
- If a plugin has bugs or is not compatible with a certain semantic-release version or use case, users will most likely open issue on the semantic-release and add to the maintenance burden
- If we list a plugin, even on the "Community" list, we kind of endorse it and user will have certain expectation
- Having a long list of plugin for the sake of it doesn't bring much value and would confuse users more than anything else

Long story-short, in the case of semantic-release, the most helpful for the community is to have a curated list of good quality, well documented, well maintained plugins. Basically quality over quantity.

**Pull request merged**